### PR TITLE
A lot of small improvements, and a fix for #86 and #89

### LINKED ISSUE #86: Placing Angel Block against a block places two angel blocks.
Steps to reproduce:
1. Have at least 2 Angel Blocks in your inventory
2. Try placing one against a further block, it will spawn two, one at the targeted point and one in front of you

Expected behaviour:
Only place one (not sure which one would that be, probably the one in front of you)

### PR DESCRIPTION
**Pull request by GallowsDove**
I've tried to fix the inconsistent use of Nonnull, NotNull and ParametersAreNonnullByDefault.
I've also done some tiny edits where I saw fit.
Furthemore, Item and Block handlers were moved outside of constructors into preRegister function, as is recommended by Slimefun JavaDocs.

At last I've fixed #86 and #89 bugs.

Needs just a bit of testing to make sure I didn't break anything,

### CODE REVIEW

File: src/main/java/io/github/sefiraat/crystamaehistoria/slimefun/tools/artistic/PoseCloner.java [hunk 1]
```diff
@@ -18,14 +18,24 @@
...
+    @Override
+    public void preRegister() {
+        super.preRegister();
+        addItemHandler(onBlockPlace());
+    }
+
+    private BlockPlaceHandler onBlockPlace() {
+        return new BlockPlaceHandler(false) {
             @Override
             public void onPlayerPlace(@NotNull BlockPlaceEvent e) {
```
Comment (Sefiraat) [inline src/main/java/io/github/sefiraat/crystamaehistoria/slimefun/tools/artistic/PoseCloner.java:32]:
```suggestion
            public void onPlayerPlace(@Nonnull BlockPlaceEvent e) {
```

File: src/main/java/io/github/sefiraat/crystamaehistoria/slimefun/tools/covers/BlockVeil.java [hunk 2]
```diff
@@ -26,11 +40,11 @@ public BlockVeil(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeT
...
                 }
 
-                for (Class<?> testClass : classToCover) {
+                for (Class<?> testClass : this.classToCover) {
                     if (testClass.isInstance(slimefunItem)) {
                         if (offhand.getType() != Material.AIR
-                            && offhand.getType().isBlock()
-                            && materialIsValid(offhand.getType())
+                                && offhand.getType().isBlock()
+                                && materialIsValid(offhand.getType())
```
Comment (Sefiraat) [inline src/main/java/io/github/sefiraat/crystamaehistoria/slimefun/tools/covers/BlockVeil.java:47]: Continuous Indent to 4

File: src/main/java/io/github/sefiraat/crystamaehistoria/utils/StoryUtils.java [hunk 3]
```diff
@@ -95,7 +96,7 @@ public static boolean isStoried(ItemStack itemStack) {
     /**
      * Sets the ItemStack's PDC Storied to True. Also sets an initial story object
      *
-     * @param itemStack The {@link ItemStack} whos meta will have the PDC element added to
+     * @param itemStack The {@link ItemStack} whose meta will have the PDC element added to
```
Comment (Sefiraat) [inline src/main/java/io/github/sefiraat/crystamaehistoria/utils/StoryUtils.java:99]: Man, I swear I got A* in English in school - I don't deserve to be trusted on peoples servers xD

**Review by Sefiraat — CHANGES_REQUESTED** [review]
It's really nice to have someone else's eye on my mess here! Thanks for this. Generally approved but I'll go through some testing when I have time today/tomorrow :)

File: src/main/java/io/github/sefiraat/crystamaehistoria/magic/spells/core/InstancePlate.java [hunk 4]
```diff
@@ -60,17 +60,14 @@ public CastResult tryCastSpell(CastInformation castInformation) {
         final Spell spell = storedSpell.getSpell();
         final int crystaCost = spell.getCrystaCost(castInformation);
 
-        // Is spell disabled in the spells.yml?
```
[the remark this answers is not in the document]
Comment (GallowsDove) [reply src/main/java/io/github/sefiraat/crystamaehistoria/magic/spells/core/InstancePlate.java:63]: Sure, it just felt like the most obvious part of the code has comments, but almost nothing else does.

**Review by Sefiraat — COMMENTED** [review]
Looking really good - I'll find some time to test in game then I'll merge :)

**Review by Sefiraat — APPROVED** [review]

**Pull request merged**